boards/riscv/adp_xc7k_ae350/adp_xc7k_ae350_defconfig

```diff
@@ -13,6 +13,10 @@ CONFIG_SERIAL=y
 # GPIO driver options
 CONFIG_GPIO=y
 
+# hwinfo driver options
+CONFIG_HWINFO=y
+CONFIG_HWINFO_ANDES=y
```

Comment:
Don't enable this option by default. See https://github.com/zephyrproject-rtos/zephyr/pull/48574.

Reply:
hwinfo is somehow different here. It does not have a device (no vtable) and therefore is optimized away by the linker if not used. The reason why it has no driver is because it cannot have more than one instance ind is always SoC internal.